packages/renderer/src/lib/kube/resource-listen.ts

```diff
@@ -0,0 +1,132 @@
+/**********************************************************************
+ * Copyright (C) 2025 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import type { CoreV1Event, KubernetesObject } from '@kubernetes/client-node';
+import type { Readable, Unsubscriber } from 'svelte/store';
+
+import type { IDisposable } from '../../../../main/src/plugin/types/disposable';
+import { isKubernetesExperimentalMode, listenResources } from './resources-listen';
+
+export interface ListenResourceOptions {
+  // plural name of the resource to listen
+  resourceName: string;
+  // name of the specific resource to listen
+  name: string;
+  // namespace to undefined to search non-namespaced resource
+  namespace?: string;
+  // set to true to handle events attached to the resource
+  listenEvents?: boolean;
+
+  // stores to be set for non-experimental mode
+  legacyResourceStore?: Readable<KubernetesObject[]>;
+  legacyEventsStore?: Readable<KubernetesObject[]>;
+
+  // called when the resource is not found
+  onResourceNotFound: () => void;
+  // called every time the resource is updated
+  onResourceUpdated: (resource: KubernetesObject, experimental: boolean) => void;
+  // called every time events attached to the resource are updated
+  onEventsUpdated?: (events: CoreV1Event[]) => void;
+}
+
+/** listenResource listens for a specific resource (resourceName, name, namespace)
+ *  and fires events when:
+ *  - the resource is not found
+ *  - the resource is updated (will send the resource without managedFields)
+ *  - the events attached to the resource are updated, optionally
+ */
+export async function listenResource(options: ListenResourceOptions): Promise<IDisposable | undefined> {
```

Comment:
Based on the usage of `listenResource` I am really wondering if we should not create a class extending the svelte/store, this is very similar, have subscribers etc.

```ts
export class ResourceStore<T> implements Readable<Array<T>> {
....
}
```

I have an example I made for the quadlet extension to contain the logs see [extension-podman-quadlet#packages/frontend/src/stores/logger-store.ts](https://github.com/podman-desktop/extension-podman-quadlet/blob/f12b86cc8fd08e425bb41a0fe5dd82d0cf2dfbf2/packages/frontend/src/stores/logger-store.ts#L16)

Reply:
Store are not deprecated, I can't find any reliable source stating this